File: app/src/main/java/com/jameschamberlain/footballteamtracker/fixtures/EditOnClickListener.java
package com.jameschamberlain.footballteamtracker.fixtures;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

public class EditOnClickListener implements View.OnClickListener {

    private Fixture fixture;
    private boolean isGoals;
    private String player;
    private RecyclerView.Adapter adapter;

    EditOnClickListener(Fixture fixture, boolean isGoals, String player, RecyclerView.Adapter adapter) {
        this.fixture = fixture;
        this.isGoals = isGoals;
        this.player = player;
        this.adapter = adapter;
    }

    @Override
    public void onClick(View v) {
        if (isGoals) {
            fixture.getGoalscorers().remove(player);
        }
        else {
            fixture.getAssists().remove(player);
        }
        adapter.notifyDataSetChanged();
    }
}

File: app/src/main/java/com/jameschamberlain/footballteamtracker/fixtures/EditOnClickListener.kt
package com.jameschamberlain.footballteamtracker.fixtures

import android.view.View
import androidx.recyclerview.widget.RecyclerView

class EditOnClickListener internal constructor(private val fixture: Fixture, private val isGoals: Boolean, private val player: String, private val adapter: RecyclerView.Adapter<*>) : View.OnClickListener {
    override fun onClick(v: View) {
        if (isGoals) {
            fixture.goalscorers.remove(player)
        } else {
            fixture.assists.remove(player)
        }
        adapter.notifyDataSetChanged()
    }

}